src/cgame/cg_weapons.cpp

```diff
@@ -2189,6 +2189,62 @@ void CG_DrawItemSelectText()
 	Rocket_SetPropertyById( "", "opacity", va( "%f", alpha ) );
 }
 
+/*
+===============
+CG_FindNextWeapon
+Find next weapon in inventory.
+===============
+*/
+weapon_t CG_FindNextWeapon( playerState_t *ps )
+{
+	weapon_t nextWeapon = WP_NONE;
+
```

Comment:
@slipher: what if we merge that working code and you submit such optimization after that? I know it may be easy to do but my brain is already full of things from that PR, I'm afraid dealing with that would be too much for me at this point (even if, taken separately, it may looks obvious). This works anyway and my brain is full.